controllers/argocd/util_test.go

```diff
@@ -544,6 +580,71 @@ func TestGetArgoApplicationControllerCommand(t *testing.T) {
 	}
 }
 
+func TestGetArgoApplicationControllerSharding(t *testing.T) {
+
+	enabled := []v1.EnvVar{
```

Comment:
Because `enabled` looks to be the same as `replicas0` and just differes from replicas3 in the value, I'd suggest factoring it out into one more simple utility function? i.e `func replicasEnvVar(value string) []v1.EnvVar { ` e.t.c.